x/cronos/keeper/keeper.go

```diff
@@ -215,3 +211,45 @@ func (k Keeper) IbcTransferCoins(ctx sdk.Context, from, destination string, coin
 	}()
 	return nil
 }
+
+// getExternalContractByDenom find the corresponding external contract for the denom,
+func (k Keeper) getExternalContractByDenom(ctx sdk.Context, denom string) (common.Address, bool) {
```

Comment:
All the methods below could be move to `params.go`

Are we not keeping the map in the paramSpace so it can be changed through governance?

Reply:
I think with paramSpace, you can only get/set param set as a whole, and the param set is a list of key-value rather than a map.
So I think we need to store in storage, only convert to/from genesis in `InitGenesis` and `ExportGenesis`, that could be done in this issue: https://github.com/crypto-org-chain/cronos/issues/46

Reply:
In bank module, the `SendEnabled` in a map<Coin, boolean> which is governed through governance, I think we can take the same design
https://github.com/cosmos/cosmos-sdk/blob/master/x/bank/types/params.go

Reply:
It depends on the number of entries in the map, because in param set, complexity of get/set operation is O(N). So unless the number of entries in the map is very low, then it's ok.

Reply:
do you mean it will be more efficient or less efficient moving it to param set?

I think we will need an entry in `params.go` anyways in order to change the map through governance

Reply:
> do you mean it will be more efficient or less efficient moving it to param set?
> 
> I think we will need an entry in `params.go` anyways in order to change the map through governance

It's much less efficient moving to param set, because when you modify it, it's always load/set as a whole, to find an entry, it's an O(N) scanning.